cos_alerter/alerter.py

```diff
@@ -151,6 +160,42 @@ def initialize():
                 "notify_time": None,
             }
 
+        # Recover any state that was dumped on last exit.
+        if config["state_file"].exists():
+            with config["state_file"].open() as f:
+                existing_clients = json.load(f)
```

Comment:
in case we may want to add more stuff to the state later on, isn't it worth it to put the existing clients in it under a `existing_clients` key instead of as the toplevel object?

Reply:
The file is called `clients.state` so I think it is good the way it is. We can always make more files in the directory. I changed the variable name to make it more clear.

Reply:
oh ok, one file per state also makes sense. didn't realize the file name corresponded to the type stored